pages/examples/components/windows.tsx

```diff
@@ -7,12 +7,113 @@ import GlobalModalManager from '@system/modals/GlobalModalManager';
+        In an organic environment, every place is unique, and the different places also cooperate, with no parts left over, to create a global whole - a whole which can be
+        identified by everyone who is part of it.
+      </div>
+    ),
+    key: 1
+  },
+  {
+    x: 634, y: 76, width: 248, height: 280, title: 'Window III',
+    children: (
+      <div style={customParagraphStyle}>
+        Making simulations of what you're going to build is tremendously useful if you can get feedback from them that will tell you where you've gone wrong and what you can do
+        about it.
+      </div>
+    ),
+    key: 2
+  },
+  {
+    x: 76, y: 394, width: 248, height: 280, title: 'Window IV',
+    children: (
+      <div style={customParagraphStyle}>
+        To work our way towards a shared language once again, we must first learn how to discover patterns which are deep, and capable of generating life.
+      </div>
+    ),
+    key: 3
+  },
+  {
+    x: 356, y: 394, width: 248, height: 280, title: 'Window V',
+    children: (
+      <div style={customParagraphStyle}>
+        From a sequence of these individual patterns, whole buildings with the character of nature will form themselves within your thoughts, as easily as sentences.
+      </div>
+    ),
+    key: 4
+  },
+  {
+    x: 634, y: 394, width: 248, height: 280, title: 'Window VI',
+    children: (
+      <div style={customParagraphStyle}>
+        When living patterns there are in a place - a room, a building, or a town - the more it comes to life as an entirety, the more it glows, the more it has that
+        self-maintaining fire which is the quality without a name.
+      </div>
+    ),
+    key: 5
+  }
+];
 
 function ExampleWindows(props) {
-  const customParagraphStyle = { fontFamily: 'var(--font-family-serif)', fontSize: '16px', lineHeight: '1.5', margin: 24 };
+  const [windows, setWindows] = React.useState<WindowProps[]>(initialWindows.map(window => ({...window, onDelete: () => closeWindow(window.key)})));
+  const [id, setId] = React.useState<number>(6);
+
+  const closeWindow = (key: number) => {
+    setWindows(prevWindows => 
+      prevWindows
+        .filter(window => window.key !== key)
+    );
+  }
+
+  function createWindow(x: number, y: number, width: number, height: number, title: string, children: React.ReactNode) {
+    const newId = id + 1;
+    setId(newId);
+
+    let newWindow: WindowProps = ({ x: x, y: y, width: width, height: height, title: title, children: children, key: newId, onDelete: () => closeWindow(newId) });
+
+    while (windows.find(window => window.x === newWindow.x && window.y === newWindow.y)) {
+      newWindow.y += 16;
+    }
+
+    setWindows([...windows, newWindow]);
+  }
+
+  function createWindows(windowProps: WindowProps[]) {
+    let newWindows = windows.concat(windowProps.map(window => ({...window, onDelete: () => closeWindow(window.key)})));
+
+    setWindows(newWindows);
+    setId(windows.length + windowProps.length);
```

Comment:
on checking back through this, i'm noticing this could lead to duplicate IDs if a window is deleted and more are added. if 8 windows with ids 0 through 7 exist, and then 5 is deleted, and 1 more is added, there will now be 2 windows with id 7.

a simpler solution that avoids all this potential conflict points might be just to generate a random number and check for duplicate before inserting.

you could add a function like "newWindowId(existing: WindowProps[])" that always generates a unique id and can avoid duplicates. you could even change the implementation later if you wanted, since it's isolated into a single function